core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -294,17 +294,27 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
 
                 futures.put(topicIdPartition, future);
             } else {
-                futures.put(topicIdPartition, CompletableFuture.completedFuture(Errors.UNKNOWN_TOPIC_OR_PARTITION));
+                futures.put(topicIdPartition, CompletableFuture.completedFuture(Errors.UNKNOWN_TOPIC_OR_PARTITION.exception()));
             }
         });
 
         CompletableFuture<Void> allFutures = CompletableFuture.allOf(
             futures.values().toArray(new CompletableFuture[0]));
         return allFutures.thenApply(v -> {
             Map<TopicIdPartition, ShareAcknowledgeResponseData.PartitionData> result = new HashMap<>();
-            futures.forEach((topicIdPartition, future) -> result.put(topicIdPartition, new ShareAcknowledgeResponseData.PartitionData()
-                .setPartitionIndex(topicIdPartition.partition())
-                .setErrorCode(future.join().code())));
+            futures.forEach((topicIdPartition, future) -> {
+                ShareAcknowledgeResponseData.PartitionData partitionData = new ShareAcknowledgeResponseData.PartitionData()
+                        .setPartitionIndex(topicIdPartition.partition());
+                Throwable t = future.join();
+                if (t == null) {
+                    partitionData.setErrorCode(Errors.NONE.code())
```

Comment:
Do we need to set the error code explicitly in case of `no error/t== null`, isn't by default the `short` value of error code should be `0`?

Reply:
For this kind of class, I’ve noticed that in some places, we check if the error is Errors.NONE.code()
https://github.com/search?q=repo%3Aapache%2Fkafka+Errors.NONE.code&type=code
I’m not sure if this is the convention, but I’ve followed the previous code and made no changes here as well. However, setErrorMessage should not be necessary for this case; I’ve removed it.